.github/workflows/ci-image-scanning.yaml

```diff
@@ -1,6 +1,11 @@
 name: image-scanning
 on:
-  push: 
+  push:
+    branches:
+      - master

```

Comment:
Maybe we can add `release-*` to here for push event.

Reply:
I probably don't think we need to scan container images in PRs. we can refer to something similar done on https://github.com/kubernetes-sigs/descheduler/pull/898/files

```yaml
name: "Security"

on:
  push:
    branches:
      - main
      - master
      - release-*
  schedule:
    - cron: '30 1 * * 0'
```

Reply:
> Maybe we can add `release-*` to here for push event.

The current strategy is that the master branch image scan is triggered when pushing, and then because there are fewer PRs for release, it is triggered by timing, referring to #4184. Of course, if you think it is necessary, we can also add `release-*` to here for push event.

Reply:
> I probably don't think we need to scan container images in PRs. we can refer to something similar done on https://github.com/kubernetes-sigs/descheduler/pull/898/files

Adding image scanning on pull_request is because we encounter a scenario. If a pr is proposed to solve a certain vulnerability, the scanning results need to be seen in PRs to determine whether the vulnerability is solved correctly.

Reply:
Thanks for the context, make sense for me.Should we only run this CI when we need to check? For example the PR has the label `area/fix-cve` or `area/security`

I tried to see how the kubernetes community does this:

1. The PR title is created in the format of [CVE-ID]
2. There is a security-related scheduled task that scans related PRs every few hours and updates the data to the dashboard, https://testgrid.k8s.io/sig-security-snyk-scan#ci-kubernetes-snyk-master

But this may not apply to karmada.

Reply:
> Thanks for the context, make sense for me.Should we only run this CI when we need to check? For example the PR has the label `area/fix-cve` or `area/security`
> 
Good idea~ Do you know how to achieve this? I didn't find the corresponding documentation.
> I tried to see how the kubernetes community does this:
> 
> 1. The PR title is created in the format of [CVE-ID]
> 2. There is a security-related scheduled task that scans related PRs every few hours and updates the data to the dashboard, https://testgrid.k8s.io/sig-security-snyk-scan#ci-kubernetes-snyk-master
> 
> But this may not apply to karmada.

Thanks for your research. I also think it makes sense to create a diff between vulnerability scan report run on master branch and the one run on HEAD (current) branch. This can be used as an optimization point for the next iteration.

Reply:
Here is a quickly written shell script.

```shell
  checklabels:
    name: check-pr-labels
    runs-on: ubuntu-22.04
    if: ${{ github.event_name == 'pull_request' }}
    outputs:
      areaSecurity: ${{ steps.checklabels.outputs.security }}
    steps:
      - name: checklabels 
        id: checklabels
        env: 
          GHEVENT: ${{ toJSON(github.event.pull_request) }}
        run: |
          if echo  $GHEVENT | jq .labels | grep -q "area/security"; then
            echo "security=true" >> "$GITHUB_OUTPUT"
          else
            echo "Label does not exist area/security"
          fi

  use-trivy-to-scan-image:
    needs: checklabels
    if: needs.checklabels.outputs.areaSecurity == 'true'
    name: image-scanning
```

Ask other people if they agree with it before start doing it.


cc  @RainbowMango @XiShanYongYe-Chang

Reply:
> and then because there are fewer PRs for release, it is triggered by timing,

Actually, CVE events will occur even if there is no PR. I will suggess to create two github action workflow,

1. schedule and push event to  run for master & release-* (today, this is release-1.6, release-1.7, release-1.8 ),
2. on pull_request to run for PR

@zhzhuang-zju  WDYH

Reply:
> Actually, CVE events will occur even if there is no PR. I will suggess to create two github action workflow,
> 
> 1. schedule and push event to  run for master & release-* (today, this is release-1.6, release-1.7, release-1.8 ),

Currently, only release 1.8 and above branches have workflow files for image scanning, so if you want to achieve this 
goal, release 1.6 and release 1.7 must also add this workflow file.

> 2. on pull_request to run for PR


Combined with the work in #4184, my current thinking is that the master branch is triggered during push and PR because it is relatively active, and the release-* branch are triggered on schedule because there are relatively few changes. We may need to find other people to decide which method is better.
cc @RainbowMango @XiShanYongYe-Chang

Reply:
The advantage of placing schedule and push with `master/release-*` in the same workflow is that you don't need to do things like `if: ${{ github.event_name == 'push' }}`  when on pull_request for PR, and the structure will be clearer.

Essentially master and release are the same thing. That's Just My Two Cents. :)

Reply:
> The advantage of placing schedule and push with `master/release-*` in the same workflow is that you don't need to do things like `if: ${{ github.event_name == 'push' }}` when on pull_request for PR, and the structure will be clearer.
> 
That's make sense for me
> Essentially master and release are the same thing, master also need to schedule scan. That's Just My Two Cents. :)

Reply:
> > and then because there are fewer PRs for release, it is triggered by timing,
> 
> Actually, CVE events will occur even if there is no PR. I will suggess to create two github action workflow,
> 
> 1. schedule and push event to  run for master & release-* (today, this is release-1.6, release-1.7, release-1.8 ),
> 2. on pull_request to run for PR
> 

Hi @liangyuanpeng, I decided to use your suggestion. Modification is ongoing, looking forward to your code review~